src/algebra/bit manipulation.md

```diff
@@ -0,0 +1,214 @@
+---
+tags:
+  - Translated
+e_maxx_link: all_submask
+---
+
+# Bit operators
+
+When finding solutions to problems, a very important factor is the simplicity and the execution time. Now, in this bitwise operations are very important and you can take advantage of them.
+
+* ## Bit shift operators
+### Multiplication and division by powers of 2
+
+There are two operators for shifting bits.
+
+**>>** One bit to the right. Which would be the same as a division by a power of two.
+
+**<<** One bit to the left. And this would be a multiplication, equally by a power of two.
+
+Let's see an example.
+
+Let's say we have a number $N$, which. $N = 4$.
+
+The binary representation of **4** is:
+
+$4_2 = 100$
+
+If we shift one bit to the left, then we would be adding a bit to the end, in this case a 0.
+
+$4_2 << 1 = 100 << 1 = 1000$
+
+$1000 = 8_2$ and clearly, $4 * 2 = 8$
+
+So, for a division by two, it would be exactly the same, but using the operator $<<$.
+
+$4_2 >> 1 = 100 >> 1 = 10$
+
+In this case, we would remove one bit from the end, this is what is called the **least significant bit**, the rightmost bit of the representation.
+
+Of course, the **most significant bit** is the leftmost bit.
+
+And then, how could it carry out operations with other powers, greater than two?
+
+Easy, you just have to remember the exponent of two, to create said power. In other words:
+
+$4 = 2 ^ 2$, so to do multiplication or division operations with **4**, we must move **2 bits**.
+
+$8 = 2 ^ 3$, so for **8** it would be **3 bits**. *And so on*.	
```

Comment:
That bit not really understandable. You should write down the syntax somewhere $x \ll y = x \cdot 2^y$.
It's not clear that you have to write $x \ll 2$ if you want to multiply by 4.

And also, this might be combined right at the beginning of this section. I don't think there a need to handle the case $y=1$ as a special case.